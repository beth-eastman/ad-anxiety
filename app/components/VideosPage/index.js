/**
 * VideoPage contains a list of thumbnails & titles to videos the user can view.
 * Selecting a video in the list will direct the user to the page containg
 * that video.
 */
import React from 'react';
import { GridList, GridTile } from 'material-ui';
import { browserHistory } from 'react-router';
import { videos } from 'data/videos';

const styles = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  gridList: {
    overflowY: 'auto',
  },
};

// Get number of columns needed for responsive grid list
const getCols = (width) => {
  if (!width) {
    return 1;
  } else if (width > 1280) {
    return 4;
  } else if (width > 900) {
    return 3;
  } else if (width > 600) {
    return 2;
  }
  return 1;
};

export default class VideosPage extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      cols: getCols(window.innerWidth),
    };
  }

  /* Change the AppBar title */
  componentWillMount() {
    this.props.setPageTitle("Videos"); // eslint-disable-line
    this.props.changeIcon('Home'); // eslint-disable-line

    // Add event listener for number of columns when window resizes
    const that = this;
    window.addEventListener('resize', function () {               // eslint-disable-line
      that.setState({ cols: getCols(window.innerWidth) });
    });
  }

  /* Render a list of all videos contaning an image and video title */
  render() {
    return (
      <div style={styles.root}>
        <GridList
          cols={this.state.cols}
          style={styles.gridList}
        >
          {videos.map((video) => (
            <GridTile
              key={video.id}
              title={video.title}
              actionPosition="right"
              onTouchTap={() => {
                browserHistory.push('/videos/'.concat(video.id));
              }}
            >
              <img src={video.poster} alt={video.title} />
            </GridTile>
          ))}
        </GridList>
      </div>
    );
  }
}
