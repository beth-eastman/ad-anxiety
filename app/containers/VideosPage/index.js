/* eslint-disable */

/**
 * Contains video contents of the application.
 */

import React from 'react';
import { GridList, GridTile } from 'material-ui';
import { browserHistory } from 'react-router';

// images and icons
import UnderstandingAnxiety from 'components/img/anxiety-understanding-anxiety.jpg';
import AnxietyTreatment from 'components/img/anxiety-types-of-anxiety.jpg';
import TreatmentOptions from 'components/img/anxiety-treatment-options.jpg';
import RecognizingTrigers from 'components/img/anxiety-recognizing-triggers.jpg';
import BackInControl from 'components/img/anxiety-back-in-control.jpg';

const styles = {
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  gridList: {
    overflowY: 'auto',
  },
};

const videos = [
  {
    id: 'understand-anxiety',
    title: 'Understanding Anxiety',
    src: UnderstandingAnxiety,
  },
  {
    id: 'types-of-anxiety',
    title: 'Types of Anxiety',
    src: AnxietyTreatment,
  },
  {
    id: 'recognizing-triggers',
    title: 'Recognizing Triggers',
    src: RecognizingTrigers,
  },
  {
    id: 'treatment-options',
    title: 'Treatment Options',
    src: TreatmentOptions,
  },
  {
    id: 'back-in-control',
    title: 'Back In Control',
    src: BackInControl,
  },
  {
    id: 'reduce-stress',
    title: 'Reduce Stress',
    src: 'https://brightcove04pmdo-a.akamaihd.net/923136676001/923136676001_3682877795001_vs-53c95f12e4b0afffb80dc10d-782203291001.jpg?pubId=923136676001&videoId=3682799653001',
  },
];

export default class VideosPage extends React.Component {

  componentWillMount() {
    this.props.setPageTitle("Videos"); // eslint-disable-line
  }

  render() {
    return (
      <div style={styles.root}>
        <GridList
          cols={1}
          style={styles.gridList}
        >
          {videos.map((tile) => (
            <GridTile
              key={tile.title}
              title={tile.title}
              actionPosition="right"
              onTouchTap={() => {
                browserHistory.push('/videos/'.concat(tile.id));
              }}
            >
              <img src={tile.src} alt={tile.title} />
            </GridTile>
          ))}
        </GridList>
      </div>
    );
  }
}
