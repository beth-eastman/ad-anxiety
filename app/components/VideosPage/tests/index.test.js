/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import VideosPage from '../index';

it('renders VideosPage', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <VideosPage setPageTitle={function () {}} changeIcon={function () {}} />
    </MuiThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
