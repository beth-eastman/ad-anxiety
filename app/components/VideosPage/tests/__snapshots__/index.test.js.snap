exports[`test renders VideosPage 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
    }
  }>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "margin": -2,
        "muiPrepared": true,
        "overflowY": "auto",
      }
    }>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Understanding Anxiety"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Understanding Anxiety
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Types of Anxiety"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Types of Anxiety
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Recognizing Triggers"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Recognizing Triggers
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Treatment Options"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Treatment Options
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Back In Control"
          src="IMAGE_MOCK"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Back In Control
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "height": 184,
          "muiPrepared": true,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }>
      <div
        cols={1}
        onTouchTap={[Function]}
        rows={1}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "muiPrepared": true,
            "overflow": "hidden",
            "position": "relative",
          }
        }>
        <img
          alt="Reduce Stress"
          src="https://brightcove04pmdo-a.akamaihd.net/923136676001/923136676001_3682877795001_vs-53c95f12e4b0afffb80dc10d-782203291001.jpg?pubId=923136676001&videoId=3682799653001"
          style={
            Object {
              "height": "100%",
              "left": "50%",
              "muiPrepared": true,
              "position": "relative",
              "transform": "translateX(-50%)",
            }
          } />
        <div
          style={
            Object {
              "alignItems": "center",
              "background": "rgba(0, 0, 0, 0.4)",
              "bottom": 0,
              "display": "flex",
              "height": 48,
              "left": 0,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
            }
          }>
          <div
            style={
              Object {
                "color": "#ffffff",
                "flexGrow": 1,
                "marginLeft": 16,
                "marginRight": 0,
                "muiPrepared": true,
                "overflow": "hidden",
              }
            }>
            <div
              style={
                Object {
                  "fontSize": "16px",
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }>
              Reduce Stress
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
